### PR TITLE
feat: add regional and edge support

### PR DESCRIPTION
**Pull request by childish-sambino**

### CODE REVIEW

File: src/Twilio/Clients/TwilioRestClient.cs [hunk 1]
```diff
@@ -35,6 +35,11 @@ public class TwilioRestClient : ITwilioRestClient
         /// </summary>
         public string Region { get; }
```
Comment (eshanholtz) [inline src/Twilio/Clients/TwilioRestClient.cs:36]: Should this also be `{ get; set; }` like Edge?

Comment (childish-sambino) [reply src/Twilio/Clients/TwilioRestClient.cs:36]: The idea is since region is tied to auth, so you shouldn't be updating it standalone. But edge is fine.

Comment (eshanholtz) [reply src/Twilio/Clients/TwilioRestClient.cs:36]: Ah, right.

File: src/Twilio/Base/Page.cs [hunk 2]
```diff
@@ -78,9 +69,9 @@ private static string UrlFromUri(Domain domain, string region, string uri)
         /// <param name="domain">Twilio subdomain</param>
         /// <param name="region">Twilio region</param>
         /// <returns>URL for the first page of results</returns>
-        public string GetFirstPageUrl(Domain domain, string region)
+        public string GetFirstPageUrl(Domain domain, string region = null)
```
Comment (eshanholtz) [inline src/Twilio/Base/Page.cs:72]: is this just for backwards compatibility? Don't see region being used in this func.

File: src/Twilio/Http/Request.cs [hunk 3]
```diff
@@ -49,9 +64,9 @@ public class Request
         public Request(HttpMethod method, string url)
         {
             Method = method;
-            _uri = new Uri(url);
-            _queryParams = new List<KeyValuePair<string, string>>();
-            _postParams = new List<KeyValuePair<string, string>>();
+            Uri = new Uri(url);
+            QueryParams = new List<KeyValuePair<string, string>>();
+            PostParams = new List<KeyValuePair<string, string>>();
```
Comment (eshanholtz) [inline src/Twilio/Http/Request.cs:69]: Why the rename?

File: src/Twilio/Base/Page.cs (hunk 2 above)
Comment (childish-sambino) [reply src/Twilio/Base/Page.cs:72]: Yes, it will be dropped soon but requires a yoyo change.

File: src/Twilio/Http/Request.cs (hunk 3 above)
Comment (childish-sambino) [reply src/Twilio/Http/Request.cs:69]: Fits more with convention since they are now auto-properties.

File: test/Twilio.Test/Http/RequestTest.cs [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+using Twilio.Http;
+using Twilio.Rest;
+
+namespace Twilio.Tests.Http
+{
+    [TestFixture]
+    public class RequestTest
+    {
+        [Test]
+        public void TestNoEdgeOrRegionInUrl()
```
Comment (eshanholtz) [inline test/Twilio.Test/Http/RequestTest.cs:12]: Really like the way you organized these

**Pull request merged**